hypothesis-python/tests/array_api/test_scalar_dtypes.py

```diff
@@ -52,6 +61,17 @@ def test_can_generate_floating_dtypes(xp, xps):
     assert_all_examples(xps.floating_dtypes(), lambda dtype: dtype in float_dtypes)
 
 
+def test_can_generate_real_dtypes(xp, xps):
+    real_dtypes = [getattr(xp, name) for name in REAL_NAMES]
+    assert_all_examples(xps.real_dtypes(), lambda dtype: dtype in real_dtypes)
```

Comment:
Good test, though misnamed - this should be renamed `test_all_generated_dtypes_are_real` and you should write an additional `test_can_generate_every_real_dtype` with pytest parametrize and find_any.

Reply:
The file actually had tests like this for all "kinds" (bool/int/uint/float/complex) and "groupy" (scalar/numeric) dtype strategies, so I ended up refactoring the lot... except for the `test_all_generated_{scalar/numeric}_dtypes_are_{scalar/numeric}`. Seemed like the right approach, but might be nicer just not change test behaviour on runtime and have say a `xp_MAX_version()` marker or something so we can test `{scalar/numeric}_dtypes()` in our generalised `test_all_generated_dtypes_are_of_group` test.